packages/jsActions/native-mobile-resources/src/camera/TakePicture.ts

```diff
@@ -50,117 +109,199 @@ export async function TakePicture(
-                    if (!unhandledError) {
-                        return resolve();
-                    }
-                    return reject(new Error(response.error));
-                }
+                            if (response.error) {
+                                const unhandledError = handleImagePickerV2Error(response.error);
 
-                return resolve(response.uri);
-            });
+                                if (!unhandledError) {
+                                    return resolve(undefined);
+                                }
+
+                                return reject(new Error(response.error));
+                            }
+
+                            return resolve(response.uri);
+                        }
+
+                        response = response as ImagePickerResponse;
+
+                        if (response.errorCode) {
+                            handleImagePickerV4Error(response.errorCode, response.errorMessage);
+
+                            return resolve(undefined);
+                        }
+
+                        return resolve(response.assets[0].uri);
+                    })
+                )
+                .catch(error => reject(error));
         });
     }
 
     function storeFile(imageObject: mendix.lib.MxObject, uri: string): Promise<boolean> {
         return new Promise((resolve, reject) => {
             fetch(uri)
-                .then(res => res.blob())
+                .then(response => response.blob())
                 .then(blob => {
-                    const guid = imageObject.getGuid();
                     // eslint-disable-next-line no-useless-escape
                     const filename = /[^\/]*$/.exec(uri)![0];
-                    const onSuccess = (): void => {
-                        NativeModules.NativeFsModule.remove(uri).then(() => {
+
+                    mx.data.saveDocument(
+                        imageObject.getGuid(),
+                        filename,
+                        {},
+                        blob,
+                        async () => {
+                            await NativeModules.NativeFsModule.remove(uri);
+
                             imageObject.set("Name", filename);
+
                             mx.data.commit({
                                 mxobj: imageObject,
                                 callback: () => resolve(true),
                                 error: (error: Error) => reject(error)
                             });
-                        });
-                    };
-                    const onError = (error: Error): void => {
-                        NativeModules.NativeFsModule.remove(uri).then(undefined);
-                        reject(error);
-                    };
-
-                    mx.data.saveDocument(guid, filename, {}, blob, onSuccess, onError);
-                });
+                        },
+                        (error: Error) => {
```

Comment:
In `takePictureAdvanced.ts`, this function is async. Why the difference?

Reply:
I'm not sure why it wasn't aligned. But looking at the bridge it returns a promise so I'll update this code.